File: android/services-v73/src/main/java/com/liferay/mobile/android/v73/source/SourceService.java
/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

package com.liferay.mobile.android.v73.source;

import com.liferay.mobile.android.service.BaseService;
import com.liferay.mobile.android.service.JSONObjectWrapper;
import com.liferay.mobile.android.service.Session;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

/**
 * @author Bruno Farache
 */
public class SourceService extends BaseService {

	public SourceService(Session session) {
		super(session);
	}

	public JSONObject addSource(long groupId, JSONObject nameMap, String driverClassName, String driverUrl, String driverUserName, String driverPassword, JSONObjectWrapper serviceContext) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("driverClassName", checkNull(driverClassName));
			_params.put("driverPassword", checkNull(driverPassword));
			_params.put("driverUrl", checkNull(driverUrl));
			_params.put("driverUserName", checkNull(driverUserName));
			_params.put("groupId", groupId);
			_params.put("nameMap", checkNull(nameMap));
			mangleWrapper(_params, "serviceContext", "com.liferay.portal.kernel.service.ServiceContext", serviceContext);

			_command.put("/reports.source/add-source", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public JSONObject deleteSource(long sourceId) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("sourceId", sourceId);

			_command.put("/reports.source/delete-source", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public JSONObject getSource(long sourceId) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("sourceId", sourceId);

			_command.put("/reports.source/get-source", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public JSONArray getSources(long groupId, String name, String driverUrl, boolean andSearch, int start, int end, JSONObjectWrapper orderByComparator) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("andSearch", andSearch);
			_params.put("driverUrl", checkNull(driverUrl));
			_params.put("end", end);
			_params.put("groupId", groupId);
			_params.put("name", checkNull(name));
			_params.put("start", start);
			mangleWrapper(_params, "orderByComparator", "com.liferay.portal.kernel.util.OrderByComparator<com.liferay.portal.reports.engine.console.model.Source>", orderByComparator);

			_command.put("/reports.source/get-sources", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONArray(0);
	}

	public Integer getSourcesCount(long groupId, String name, String driverUrl, boolean andSearch) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("andSearch", andSearch);
			_params.put("driverUrl", checkNull(driverUrl));
			_params.put("groupId", groupId);
			_params.put("name", checkNull(name));

			_command.put("/reports.source/get-sources-count", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getInt(0);
	}

	public JSONObject updateSource(long sourceId, JSONObject nameMap, String driverClassName, String driverUrl, String driverUserName, String driverPassword, JSONObjectWrapper serviceContext) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("driverClassName", checkNull(driverClassName));
			_params.put("driverPassword", checkNull(driverPassword));
			_params.put("driverUrl", checkNull(driverUrl));
			_params.put("driverUserName", checkNull(driverUserName));
			_params.put("nameMap", checkNull(nameMap));
			_params.put("sourceId", sourceId);
			mangleWrapper(_params, "serviceContext", "com.liferay.portal.kernel.service.ServiceContext", serviceContext);

			_command.put("/reports.source/update-source", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

}

File: windows/Liferay.SDK/Service/V73/Source/SourceService.cs
/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Threading.Tasks;

namespace Liferay.SDK.Service.V73.Source
{
	public class SourceService : ServiceBase
	{
		public SourceService(ISession session)
			: base(session)
		{
		}

		public async Task<IEnumerable<dynamic>> GetSourcesAsync(long groupId, string name, string driverUrl, bool andSearch, int start, int end, JsonObjectWrapper orderByComparator)
		{
			var _parameters = new JsonObject();

			_parameters.Add("groupId", groupId);
			_parameters.Add("name", name);
			_parameters.Add("driverUrl", driverUrl);
			_parameters.Add("andSearch", andSearch);
			_parameters.Add("start", start);
			_parameters.Add("end", end);
			this.MangleWrapper(_parameters, "orderByComparator", "$languageUtil.getJSONWrapperClassName($parameter.type)", orderByComparator);

			var _command = new JsonObject()
			{
				{ "/reports.source/get-sources", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (IEnumerable<dynamic>)_obj;
		}

		public async Task<dynamic> UpdateSourceAsync(long sourceId, IDictionary<string, string> nameMap, string driverClassName, string driverUrl, string driverUserName, string driverPassword, JsonObjectWrapper serviceContext)
		{
			var _parameters = new JsonObject();

			_parameters.Add("sourceId", sourceId);
			_parameters.Add("nameMap", nameMap);
			_parameters.Add("driverClassName", driverClassName);
			_parameters.Add("driverUrl", driverUrl);
			_parameters.Add("driverUserName", driverUserName);
			_parameters.Add("driverPassword", driverPassword);
			this.MangleWrapper(_parameters, "serviceContext", "$languageUtil.getJSONWrapperClassName($parameter.type)", serviceContext);

			var _command = new JsonObject()
			{
				{ "/reports.source/update-source", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<dynamic> AddSourceAsync(long groupId, IDictionary<string, string> nameMap, string driverClassName, string driverUrl, string driverUserName, string driverPassword, JsonObjectWrapper serviceContext)
		{
			var _parameters = new JsonObject();

			_parameters.Add("groupId", groupId);
			_parameters.Add("nameMap", nameMap);
			_parameters.Add("driverClassName", driverClassName);
			_parameters.Add("driverUrl", driverUrl);
			_parameters.Add("driverUserName", driverUserName);
			_parameters.Add("driverPassword", driverPassword);
			this.MangleWrapper(_parameters, "serviceContext", "$languageUtil.getJSONWrapperClassName($parameter.type)", serviceContext);

			var _command = new JsonObject()
			{
				{ "/reports.source/add-source", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<long> GetSourcesCountAsync(long groupId, string name, string driverUrl, bool andSearch)
		{
			var _parameters = new JsonObject();

			_parameters.Add("groupId", groupId);
			_parameters.Add("name", name);
			_parameters.Add("driverUrl", driverUrl);
			_parameters.Add("andSearch", andSearch);

			var _command = new JsonObject()
			{
				{ "/reports.source/get-sources-count", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (long)_obj;
		}

		public async Task<dynamic> GetSourceAsync(long sourceId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("sourceId", sourceId);

			var _command = new JsonObject()
			{
				{ "/reports.source/get-source", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<dynamic> DeleteSourceAsync(long sourceId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("sourceId", sourceId);

			var _command = new JsonObject()
			{
				{ "/reports.source/delete-source", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}
	}
}